fix(RecentNotes): Prevent folder pages from always appearing first (closes #1901)

kaygdotorg opened the pull request:
Pass prioritizeFolders=false to byDateAndAlphabetical in RecentNotes to sort strictly by date/alphabetical order, fixing issue #1901.

kaygdotorg committed:
Fix(RecentNotes): Prevent folder pages from always appearing first

Pass prioritizeFolders=false to byDateAndAlphabetical in RecentNotes to sort strictly by date/alphabetical order, fixing issue #1901.

jackyzha0 reviewed (commented):
just split out two separate sorts, one for byDateAndAlphabetical and another for byDateAndAlphabeticalFolderFirst

kaygdotorg committed:
refactor: split sorting functions for clarity

- Split byDateAndAlphabetical into two separate functions\n- byDateAndAlphabetical: sorts strictly by date and alphabetically\n- byDateAndAlphabeticalFolderFirst: sorts with folders first\n- Updated RecentNotes to use date-only sorting

kaygdotorg commented:
Does it look fine now?

jackyzha0 reviewed (commented):
can you make sure the previous cases the used byDateAndAlphabetical now use byDateAndAlphabeticalFolderFirst?

kaygdotorg commented:
I made `byDateAndAlphabetical` the default as I thought it'd be the most logical thing to do. But I can change it to `byDateAndAlphabeticalFolderFirst` in PageList, np. That's what you mean, right?

jackyzha0 commented:
ah yes thats what i meant, sorry!

kaygdotorg committed:
Fix(PageList): keep byDateAndAlphabeticalFolderFirst as the default sorting order for PageList

kaygdotorg commented:
Done.

jackyzha0 commented:
thanks!
